pkg/restore/import.go

```diff
@@ -333,13 +296,29 @@ func (importer *FileImporter) ingestSST(
 	}
 	respErr := resp.GetError()
 	if respErr != nil {
-		if respErr.EpochNotMatch != nil {
-			return errEpochNotMatch
+		log.Debug("ingest sst resp error", zap.Stringer("error", respErr))
+		if respErr.GetKeyNotInRegion() != nil {
+			return errKeyNotInRegion
 		}
-		if respErr.NotLeader != nil {
+		if respErr.GetNotLeader() != nil {
 			return errNotLeader
 		}
-		return errors.Errorf("ingest failed: %v", respErr)
+		return errResp
 	}
 	return nil
 }
+
+func extractDownloadSSTError(e error) error {
+	err := errGrpc
+	switch {
+	case strings.Contains(e.Error(), "bad format"):
+		err = errBadFormat
+	case strings.Contains(e.Error(), "wrong prefix"):
+		err = errWrongKeyPrefix
+	case strings.Contains(e.Error(), "corrupted"):
+		err = errFileCorrupted
+	case strings.Contains(e.Error(), "Cannot read"):
+		err = errCannotRead
```

Comment:
Could we define these errors in protobuf? @kennytm

Reply:
Frankly it's ridiculous that we return a structured error in Rust and it got serialized into *debug* representation. What if we change https://github.com/tikv/tikv/blob/3ba72ab98ed8f3155bca4bd8fed13d1ddbf1a02a/components/sst_importer/src/service.rs#L7-L10 and return a structured (JSON) message?

Reply:
I guess that is because we did not define the structured error in protobuf.

In our use case, I think we should choose "Richer error model" https://grpc.io/docs/guides/error/

> and return a structured (JSON) message?

IMO, we should use one data format to avoid confusion, that is protobuf.

Reply:
@overvenus Thanks for the pointer. The Rust `grpcio` package we're using [doesn't have that `repeated Any details = 3` field](https://github.com/tikv/grpc-rs/blob/35564e56f0becb3f014dee0c023717958e768f4f/src/call/mod.rs#L133-L141) described in https://cloud.google.com/apis/design/errors#error_model though.  So we need to first update `grpcio` before using the rich model.

Reply:
Another way is including error message in response.

Anyway, it should be future work, let's focus on enhance error handling.